src/Microsoft.AspNetCore.Http.Abstractions/Extensions/IEndpointModelConvention.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Builder
+{
+    /// <summary>
+    /// Allows customization of the <see cref="EndpointModel"/>.
+    /// </summary>
+    /// <remarks>
+    /// This interface is used at application startup to customize endpoints for the application.
+    /// </remarks>
+    public interface IEndpointConventionBuilder
```

Comment:
IEndpointConvention~~Builder~~?

Reply:
Yes the action is the convention. Example of it being used:

```cs
public static class EndpointConventionBuilderExtensions
{
    public static IEndpointConventionBuilder RequireAuthorization(this IEndpointConventionBuilder builder, params string[] roles)
    {
        builder.Apply(endpoint => endpoint.Metadata.Add(new AuthorizationMetadata(roles)));
        return builder;
    }
}
```

Sure you could replace the action with an interface. I don't see any benefit from that though, and creating a new type and having it keep state has a lot of overhead compared to a lambda + closure. Right now it is very easy to create an extension method (or use Apply directly when building endpoints) to customize endpoints.